torch/testing/_internal/common_utils.py

```diff
@@ -927,6 +927,34 @@ def get_comparison_dtype(a, b):
 
     return compare_dtype
 
+def compare_with_reference(torch_fn, ref_fn, sample_input):
+    # Compares torch function with reference function for given sample input (object of SampleInput)
+    # Note: only values are compares, type comparison is not done here
+    try:
+        n_inp, n_args, _ = sample_input.numpy()
+    except TypeError:
+        return
+    t_inp, t_args, _ = sample_input.splat()
```

Comment:
The output from `splat()` needs to be explicitly splatted like `torch_fn(t_input, *t_args, **t_kwargs)` else following error is raised if used as: `torch_fn(`sample_input.splat())`:

```
======================================================================
ERROR: test_reference_testing_sub_test_subtract_cpu_int64 (__main__.TestOpInfoCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
    result = test_fn(self, *args)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 266, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 627, in wrapper
    fn(*args, **kwargs)
  File "/home/krshrimali/git/krshrimali/pytorch/test/test_ops.py", line 149, in test_reference_testing
    compare_with_reference(op, op.ref, sample_input)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_utils.py", line 942, in compare_with_reference
    torch_fn(*sample_input.splat())
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_methods_invocations.py", line 331, in __call__
    return self.op(*args, **kwargs)
TypeError: sub() takes 2 positional arguments but 3 were given

----------------------------------------------------------------------
```